File: datetime/easter/easter.erl
-module(easter).
-export([main/0]).

get_easter_for(Y) ->
    G = (Y  rem 19) + 1,
    C = (Y div 100) + 1,
    X = (3 * C div 4) - 12,
    Z = (((8 * C) + 5) div 25) - 5,
    D = ((5 * Y) div 4) - X - 10,
    Et = ((11 * G) + 20 + Z - X) rem 30,
    E =  if
        ((Et == 25) and (G > 11)) or (Et == 24) -> Et + 1;
        true -> Et
    end,
    Nfmt = 44 - E,
    Nfm = if
        Nfmt < 21 -> Nfmt + 30;
        true -> Nfmt
    end,
    N = Nfm + 7 - ((D + Nfm) rem 7),
    Month = if
        N > 31 -> "April";
        true -> "March"
    end,
    Day = if
        N > 31 -> N - 31;
        true -> N
    end,
    {Y, Month, Day}.

get_easters(Year, EndYear, Pid) when Year > EndYear ->
    Pid ! {stop};
get_easters(Year, EndYear, Pid) ->
    Pid ! {easter, get_easter_for(Year)},
    get_easters(Year + 1, EndYear, Pid).

print_easters_loop(From) ->
    receive
        {easter, Easter} ->
            io:format("   ~02B ~s, ~04B\n",
                [element(3, Easter), element(2, Easter), element(1, Easter)]),
            print_easters_loop(From);
        {stop} ->
            From ! {stop}
    end.
print_easters(From) ->
    io:format("Easters:~n"),
    print_easters_loop(From).

main() ->
    PrimaryPid = self(),
    Printer = spawn(fun() -> print_easters(PrimaryPid) end),
    spawn(fun() -> get_easters(1950, 2050, Printer) end),
    receive
     {stop} -> ok
    end.

File: datetime/easter/easter.exs

defmodule Easters do
  def get_easter_for(y) do
    g = rem(y, 19) + 1
    c = div(y, 100) + 1
    x = div(3 * c, 4) - 12
    z = div((8 * c) + 5, 25) - 5
    d = div(5 * y, 4) - x - 10
    et = rem((11 * g) + 20 + z - x, 30)
    e = if ((et == 25) and (g > 11) or (et == 24)) do
      et + 1
    else
      et
    end
    nfmt = 44 - e
    nfm = if nfmt < 21 do
      nfmt + 30
    else
      nfmt
    end
    n = nfm + 7 - rem(d + nfm, 7)
    month = if n > 31 do
      "April"
    else
      "March"
    end
    day = if n > 31 do
      n - 31
    else
      n
    end
    {y, month, day}
  end

  def get_easters(year, endYear, pid) when year > endYear do
    send(pid, {:stop})
  end
  def get_easters(year, endYear, pid) do
    send(pid, {:easter, get_easter_for(year)})
    get_easters(year + 1, endYear, pid)
  end
end

defmodule EastersPrinter do
  def print_easters_loop(pid) do
    receive do
      {:easter, easter} ->
        IO.write(:stdio, :io_lib.format(
          "   ~2..0B ~s, ~4..0B~n",
          [elem(easter, 2), elem(easter, 1), elem(easter, 0)]))
          print_easters_loop(pid)
      {:stop} ->
        send(pid, {:stop})
    end
  end

  def print_easters(pid) do
    IO.puts("Easters:")
    print_easters_loop(pid)
  end
end

primaryPid = self()
printer = spawn(fn -> EastersPrinter.print_easters(primaryPid) end)
spawn(fn -> Easters.get_easters(1950, 2050, printer) end)
receive do
  {:stop} -> :ok
end
